lib/bq_stream/bq_stream.rb

```diff
@@ -14,86 +14,188 @@ module BqStream
+      start_after_id = QueuedItem.where(sent_to_bq: true).last.id || 0 # added for testing TODO: update after test
+      log_code = rand(2**256).to_s(36)[0..7]
+      log(:info, "#{Time.now}: ***** Dequeue Items Started ***** #{log_code}")
+      OldestRecord.update_bq_earliest
+      create_bq_writer
+      # records = QueuedItem.all.limit(batch_size) # removing for testing TODO: update after test
+      records = QueuedItem.where('id > ?', start_after_id).limit(batch_size)
+      log(:info, "#{Time.now}: Records Count: #{records.count} #{log_code}")
+      data = records.collect do |i|
+        new_val = encode_value(i.new_value) rescue nil
+        { table_name: i.table_name, record_id: i.record_id, attr: i.attr,
+          new_value: new_val ? new_val : i.new_value, updated_at: i.updated_at }
       end
-    end if old_records['rows']
-    # logger.info "#{Time.now}: $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$ End Initialize Old Records $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$"
-  end
+      @bq_writer.insert(bq_table_name, data) unless data.empty?
+      records.each { |r| r.update(sent_to_bq: true) } # added for testing TODO: update after test
+      # QueuedItem.delete_all_with_limit # removing delete for testing TODO: reinstate after test
+      log(:info, "#{Time.now}: ***** Dequeue Items Ended ***** #{log_code}")
+    end
 
-  def self.encode_value(value)
-    value.encode('utf-8', invalid: :replace,
-                          undef: :replace, replace: '_') rescue nil
-  end
+    def create_bq_dataset
+      @bq_writer.create_dataset(dataset)
+    end
 
-  def self.dequeue_items
-    # log_code = rand(2**256).to_s(36)[0..7]
-    # logger.info "#{Time.now}: ***** Dequeue Items Started ***** #{log_code}"
-    OldestRecord.update_bq_earliest
-    create_bq_writer
-    records = QueuedItem.all.limit(batch_size)
-    # logger.info "#{Time.now}: Records Count: #{records.count} #{log_code}"
-    data = records.collect do |i|
-      new_val = encode_value(i.new_value) rescue nil
-      { table_name: i.table_name, record_id: i.record_id, attr: i.attr,
-        new_value: new_val ? new_val : i.new_value, updated_at: i.updated_at }
-    end
-    @bq_writer.insert(bq_table_name, data) unless data.empty?
-    QueuedItem.delete_all_with_limit
-    # logger.info "#{Time.now}: ***** Dequeue Items Ended ***** #{log_code}"
-  end
+    def create_bq_table
+      bq_table_schema = { table_name:   { type: 'STRING', mode: 'REQUIRED' },
+                          record_id:    { type: 'INTEGER', mode: 'REQUIRED' },
+                          attr:         { type: 'STRING', mode: 'NULLABLE' },
+                          new_value:    { type: 'STRING', mode: 'NULLABLE' },
+                          updated_at:   { type: 'TIMESTAMP', mode: 'REQUIRED' } }
+      @bq_writer.create_table(bq_table_name, bq_table_schema)
+    end
 
-  def self.create_bq_dataset
-    @bq_writer.create_dataset(dataset)
-  end
+    def big_query_check(klass, dataset, table, qty)
```

Comment:
This class is getting pretty big, plus all this code is specific to your new check system. I think it would be nice to move it into a different class or module to keep things organized.